flow/util/misc.py

```diff
@@ -199,3 +200,19 @@ def keys_used(self):
 def fullmatch(regex, string, flags=0):
     """Emulate python-3.4 re.fullmatch()."""
     return re.match("(?:" + regex + r")\Z", string, flags=flags)
+
+
+def roundrobin(*iterables):
+    # From: https://docs.python.org/3/library/itertools.html#itertools-recipes
+    # roundrobin('ABC', 'D', 'EF') --> A D E B F C
+    # Recipe credited to George Sakkis
+    num_active = len(iterables)
+    nexts = cycle(iter(it).__next__ for it in iterables)
```

Comment:
Are you inclined to leave this recipe unchanged since it was copied? If so, that's fine, but if you're open to changes I would rewrite this as `cycle(iter(next(it)))` rather than calling `__next__`.

Reply:
Your solution would actually be wrong (I tested it). So I'm leaving as-is.

Reply:
Interesting. Do you know why that is?

Reply:
It is because we are creating a cycle of `__next__` functions, not a cycle of *next* elements.